climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1158 @@
+
+            firms_cons = firms_iter[firms_iter.event_id == event_id]
+
+            # Creation of an identifier for geographical clustering
+            # For each temporal cluster, perform geographical clustering with DBSCAN algo
+            for cons_id in np.unique(firms_cons['cons_id'].values):
+                temp = np.argwhere(firms_cons['cons_id'].values == cons_id).reshape(-1,)
+                lat_lon = firms_cons.iloc[temp][['latitude', 'longitude']].values
+                lat_lon_uni, lat_lon_cpy = np.unique(lat_lon, return_inverse=True, axis=0)
+                cluster_id = DBSCAN(eps=res_data*clus_thres, min_samples=1).\
+                                    fit(lat_lon_uni).labels_
+                cluster_id = cluster_id[lat_lon_cpy]
+                firms_cons.clus_id.values[temp] = cluster_id
+                firms_iter.clus_id.values[firms_iter.event_id == event_id] = \
+                    firms_cons.clus_id.values
+
+        firms.clus_id.values[firms['iter_ev'].values] = firms_iter.clus_id.values
+
+        return firms
+
+    @staticmethod
+    @numba.njit(parallel=True)
+    def _firms_fire(days_thres, fir_cons_id, fir_clus_id, fir_ev_id, fir_iter,
+                    fir_date):
+        """Creation of event_id for each dataset point.
+        A fire is characterized by a unique combination of 'cons_id' and 'clus_id'.
+
+        Parameters:
+            firms (dataframe)
+
+        Returns:
+            firms
+        """
+        ev_id = 0
+        for cons_id in np.unique(fir_cons_id):
+            firms_cons = fir_clus_id[fir_cons_id == cons_id]
+            for clus_id in np.unique(firms_cons):
+                fir_ev_id[np.logical_and(fir_cons_id == cons_id, \
+                fir_clus_id == clus_id)] = ev_id
+                ev_id += 1
+
+        for ev_id in np.unique(fir_ev_id):
+            date_ord = np.sort(fir_date[fir_ev_id == ev_id])
+            if (np.diff(date_ord) < days_thres).all():
+                fir_iter[fir_ev_id == ev_id] = False
+            else:
+                fir_iter[fir_ev_id == ev_id] = True
+
+    @staticmethod
+    def _firms_remove_minor_fires(firms, minor_fires_thres):
+        """Remove fires containg fewer FIRMS entries than threshold.
+        A fire is characterized by a unique combination of 'cons_id' and 'clus_id'.
+
+        Parameters:
+            firms (dataframe)
+            minor_fires_thres(int)
+
+        Returns:
+            firms
+        """
+        for i in range(np.unique(firms.event_id).size):
+            if (firms.event_id == i).sum() < minor_fires_thres:
+                firms = firms.drop(firms[firms.event_id == i].index)
+        # assign new event IDs
+        event_id_new = 1
+        for i in np.unique(firms.event_id):
+            firms.event_id[firms.event_id == i] = event_id_new
+            event_id_new = event_id_new + 1
+
+        firms = firms.reset_index()
+
+        return firms
+
+    @staticmethod
+    def _centroids_resolution(centroids):
```

Comment:
This should either be in the `Centroids` module or in the `Util` module.

Reply:
moved to Centroids